schema/configuration.go

```diff
@@ -419,48 +442,49 @@ func GetDataSharing(cfgBytes []byte) (*DataSharing, error) {
 	var cfgIface interface{}
 	err := json.Unmarshal(cfgBytes, &cfgIface)
 	if err != nil {
-		return nil, MalformedConfigError
+		return nil, MalformedConfigError{What: "JSON error"}
 	}
 
 	dataSharing := new(DataSharing)
 
 	cfg, ok := cfgIface.(map[string]interface{})
 	if !ok {
-		return dataSharing, MalformedConfigError
+		return dataSharing, MalformedConfigError{What: "Config file is empty"}
 	}
 
-	dscfg, ok := cfg["DataSharing"]
+	const DATASHARING string = "DataSharing"
```

Comment:
We usually just make constants with capital first letters instead of all caps, FYI. Might also be a good idea to group the constants at the top of the function:
```
const (
   JSON-API = "JSON-API"
   DataSharing = "DataSharing"
)
```